src/library/styles/createStyledComponent.js

```diff
@@ -1,31 +1,76 @@
 /* @flow */
-import glamorous from 'glamorous';
+
+import styled from 'react-emotion';
+import withPropsFn from 'recompose/withProps';
 import componentStyleReset from './componentStyleReset';
+import isValidProp from '../utils/isValidProp';
+import { isProduction } from '../utils/nodeEnv';
 
 export default function createStyledComponent(
   element:
     | React$StatelessFunctionalComponent<*>
     | React$ComponentType<*>
     | string,
-  styles: Object | ((props: Object, context?: Object) => Object),
-  options?: Object = {}
+  styles:
+    | Object
+    | Array<Object>
+    | ((props: Object, context?: Object) => Object | Array<Object>),
+  options?: {
+    displayName?: string,
+    filterProps?: Array<string>,
+    forwardProps?: Array<string>,
+    includeStyleReset?: boolean,
+    rootEl?: string,
+    withProps?: Object
+  } = {}
 ) {
-  const { includeStyleReset, ...restOptions } = options;
-  let outStyles;
+  const {
+    displayName,
+    filterProps = [],
+    forwardProps = [],
+    rootEl,
+    withProps
+  } = options;
+  const includeStyleReset = options.includeStyleReset || false;
 
-  if (includeStyleReset) {
-    outStyles = (props: Object, context?: Object): Object => {
-      const componentStyles =
-        typeof styles === 'function' ? styles(props, context) : styles;
+  const outStyles = (
+    props: Object,
+    context?: Object
+  ): Object | Array<Object> => {
+    let componentStyles =
+      typeof styles === 'function' ? styles(props, context) : styles;
 
-      return {
-        ...componentStyleReset(props),
+    if (Array.isArray(componentStyles)) {
+      componentStyles.unshift(componentStyleReset(props));
```

Comment:
If `styles` is an array, `componentStyleReset` is always included?

Reply:
Wops! Fixing...